docs/proposals/prevent-removal-managed-resources/README.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Prevent removal managed resources
+authors:
+- "@CharlesQQ"
+reviewers:
+- "@RainbowMango"
+- "@XiShanYongYe-Chang"
+- "@chaunceyjiang"
+approvers:
+
+creation-date: 2024-06-27
+
+# prevent removal managed resource
+
+## Summary
+
+A deletion mechanism and its API is provided to prevent cascade resource in member cluster deleted by control plane.
+
+## Motivation
+
+All resources reconciled by the execution controller have a finalizer `karmada.io/execution-controller` added to work object's metedata.
+This finalizer will prevent deletion of work object until the execution controller has a chance to perform pre-deletion cleanup.
+
+However, in some cases, user want to delete resource in control plane but resource in member cluster kept, like `kubectl delete sts xxx --cascade=orphan`, sts is deleted but pods not.
+Karmada have different types of resources, different resource might have different prevent removal process.
+
+1. resource template: clean up the labels/annotations from the resource in member clusters and delete work object.
+2. dependent resource: clean up the labels/annotations from the dependent resource in member clusters,such as configmap/secret,  and then delete work object.
+3. resource which directly generate work object such as namespace/cronfederatedhpa/federaredhpa/federatedresourcequota execute the same strategy as resource template.
```

Comment:
What does `same strategy` mean?

I think namespace is different from the other resources, I believe we need to explain namespace separately.

Not all namespaces directly generate work. They can also be matched by some PP. (For example, I have already turned off the ns-controller.)